### PR TITLE
feat: add simple random forest classifier for metal classification

### PR DESCRIPTION
**Pull request by kbiniek**
cheers

### CODE REVIEW

File: scripts/training/train_simple_RFC.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.preprocessing import LabelEncoder
+from sklearn.metrics import accuracy_score
+
+# from nmrcraft.analysis.plotting import plot_predicted_vs_ground_truth
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:5]: Super nice! We could import some plotting functions for e.g. the confusion matrix here in the future

File: scripts/training/train_simple_RFC.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+set_seed()
+
+
+def main():
+    """Train a simple random forest model."""
+    # Load the data
+    dataset = load_data()
+
+    # Only take 1% of the dataset
+    dataset = dataset.sample(frac=0.01)
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:18]: I think this is a remnant of the trial code in the RFR. We should do it a bit more modular in the future – so that we can also run multiple experiments with multiple “number of training data”. (To essetially show: more data -> better accuracy)
I checked and with using 100% of the dataset we get 95% accuracy !

File: scripts/training/train_simple_RFC.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    # Extract features and target variables
+    X = dataset[['M_sigma11_ppm', 'M_sigma22_ppm', 'M_sigma33_ppm', 'E_sigma11_ppm', 'E_sigma22_ppm', 'E_sigma33_ppm']].to_numpy()
+    y_label = dataset['metal'].to_numpy()
+    # TODO automatize feature selection with hydra for runs
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:23]: Perhaps we can get rid of those TODO comments and make Issues out of it for the ’non-experimental run'?
I can look into using hydra for tracking the model runs !

File: scripts/training/train_simple_RFC.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    # Transform the target variable using the label encoder
+    label_encoder = LabelEncoder()
+    label_encoder.fit(["Mo", "W"])
+    y = label_encoder.transform(y_label)
+    # TODO create pipeline
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:29]: with that I would mean a pipeline to have a StandardScaler, LabelEncoder etc. in a pipeline https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html we could look into this in the future

File: scripts/training/train_simple_RFC.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    # Split the data into training and testing sets
+    X_train, X_test, y_train, y_test = split_data(X, y)
+
+    # Create a random forest classifier
+    model = RandomForestClassifier()
+    # TODO optimize hyperparameters
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:36]: we can play around with the lecture hyperparam optimization methods here (possibly add some functions like NMRcraft/training/hyperparameter_optimization.py and then tun model.tune_hyperparameters())

File: scripts/training/train_simple_RFC.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    # Train the model
+    model.fit(X_train, y_train)
+
+    # Evaluate the model on the testing data
+    y_pred = model.predict(X_test)
+
+    # Calculate the accuracy score
+    score = accuracy_score(y_test, y_pred)
```
Comment (mlederbauer) [inline scripts/training/train_simple_RFC.py:45]: other suggestions (possibly add this to NMRcraft/analysis): F1score, False Positive Rate, Confusion matrix
can be added later

**Review by mlederbauer — APPROVED** [review]
I would suggest to delete some comments that are remnants of the “test random forest regressor” and add a bit more functionality for the “final run” (hyperparameter optimization, evaluation metrics, …)  super cool!

**Comment by tiaguinho-code** [comment]
Looks very nice and I agree with all the comments @mlederbauer make regarding possible features we could expand on in the future :D

**Review by tiaguinho-code — APPROVED** [review]
Looks very nice and I agree with all the comments @mlederbauer make regarding possible features we could expand on in the future :D
